c10/core/impl/LocalDispatchKeySet.h

```diff
@@ -24,67 +25,45 @@
 namespace c10 {
 namespace impl {
 
-// POD version of LocalDispatchKeySet.  Declared here just so that
-// we can put it in the guards.
-// This struct encapsulates special handling for TLS initialization
-// in set_included()/included() API so that they reflect the truth.
-// If you want to create PODLocalDispatchKeySet with non-zero state,
-// use set_included() instead of default constructor.
-struct C10_API PODLocalDispatchKeySet {
-  uint64_t included_;
-  uint64_t excluded_;
+// Helper class to handle conversion between the TLS zero initialized
+// convention and the "normal" in memory layout.
+class C10_API LocalDispatchKeySetWrapper {
+ public:
+  LocalDispatchKeySetWrapper(PODLocalState* tls) : tls_(tls) {}
+  LocalDispatchKeySetWrapper() : tls_(_get_thread_local_state()) {}
 
   // See Note [TLS Initialization]
   DispatchKeySet included() const {
-    return DispatchKeySet(DispatchKeySet::RAW, included_) ^
+    return DispatchKeySet(DispatchKeySet::RAW, tls_->included_) ^
         c10::default_included_set;
   }
   DispatchKeySet excluded() const {
-    return DispatchKeySet(DispatchKeySet::RAW, excluded_) ^
+    return DispatchKeySet(DispatchKeySet::RAW, tls_->excluded_) ^
         c10::default_excluded_set;
   }
 
-  void set_included(DispatchKeySet x) {
-    included_ = (x ^ c10::default_included_set).raw_repr();
+  void unsafe_set_included(DispatchKeySet x) {
```

Comment:
It behooves you to explain why these are unsafe

Reply:
I don't actually remember. I think it might have been from an earlier iteration where I had dropped the mask. (I'll drop the name now.)